PayPalNativeCheckout/src/main/java/com/braintreepayments/api/PayPalNativeCheckoutClient.java

```diff
@@ -1,14 +1,188 @@
+                        payPalRequest.getReturnUrl()
+                    )
+                );
+
+                registerCallbacks(configuration, payPalRequest, payPalResponse);
+
+                PayPalCheckout.startCheckout(createOrderActions -> {
+                    if (payPalRequest instanceof PayPalNativeCheckoutRequest) {
+                        createOrderActions.set(payPalResponse.getPairingId());
+                    } else if (payPalRequest instanceof PayPalNativeCheckoutVaultRequest) {
+                        createOrderActions.setBillingAgreementId(payPalResponse.getPairingId());
+                    }
+                });
+            } else {
+                throw new Exception("Checkout response was empty");
+            }
+        });
+    }
+
+    private void registerCallbacks(
+        final Configuration configuration,
+        final PayPalNativeRequest payPalRequest,
+        final PayPalNativeCheckoutResponse payPalResponse
+    ) {
+        PayPalCheckout.registerCallbacks(
+                approval -> {
+                    PayPalNativeCheckoutAccount payPalAccount = setupAccount(configuration, payPalRequest, payPalResponse);
+                    internalPayPalClient.tokenize(payPalAccount, (payPalAccountNonce, error) -> {
+                        if (payPalAccountNonce != null) {
+                            listener.onPayPalSuccess(payPalAccountNonce);
+                        } else {
+                            listener.onPayPalFailure(new Exception("PaypalAccountNonce is null"));
+                        }
+                    });
+                },
+                null,
+                () -> listener.onPayPalFailure(new Exception("User has canceled")),
+                errorInfo -> listener.onPayPalFailure(new Exception(errorInfo.getError().getMessage()))
+        );
+    }
+
+    private PayPalNativeCheckoutAccount setupAccount(
+        final Configuration configuration,
+        final PayPalNativeRequest payPalRequest,
+        final PayPalNativeCheckoutResponse payPalResponse
+    ) {
+        PayPalNativeCheckoutAccount payPalAccount = new PayPalNativeCheckoutAccount();
+
+        String merchantAccountId = payPalRequest.getMerchantAccountId();
+        String paymentType = payPalRequest instanceof PayPalNativeCheckoutVaultRequest ? "billing-agreement" : "single-payment";
+        payPalAccount.setClientMetadataId(configuration.getPayPalClientId());
+        payPalAccount.setIntent(payPalResponse.getIntent());
+        payPalAccount.setSource("paypal-browser");
+        payPalAccount.setPaymentType(paymentType);
+
+        if (merchantAccountId != null) {
+            payPalAccount.setMerchantAccountId(merchantAccountId);
+        }
+
+        return payPalAccount;
+    }
+
+    private static String getAnalyticsEventPrefix(PayPalNativeRequest request) {
+        return request instanceof PayPalNativeCheckoutVaultRequest ? "paypal.billing-agreement" : "paypal.single-payment";
```

Comment:
```suggestion
        return request instanceof PayPalNativeCheckoutVaultRequest ? "paypal-native.billing-agreement" : "paypal-native.single-payment";
```

Reply:
Done, switched events to use `paypal-native`